File: examples/actor.scala
import scala.actors.Actor

case class Mult(value: Int)
case class Add(value: Int)
case class Something(value: Int)
case object Stop

class MyActor extends Actor {
   def act() {
      loop {
         receive {
            case Mult(value) => println(value + " * " + value + " = " + value * value)
            case Add(value) => println(value + " + " + value + " = " + (value + value))
            case Stop => exit()
            case msg => println("Don't know what to do with message :" + msg)
         }
      }
   }
}

val myActor = new MyActor()
myActor.start

myActor ! Mult(4)
myActor ! Add(4)
myActor ! Something(4)

myActor ! Stop

File: examples/actor.rb
$:.unshift("../lib")
require 'ara'

class MyActor < Actor
  def receive(message)
    action = message[0]
    value = message[1]
    case action
      when :mult then
        puts "#{value} * #{value} = #{value * value}"
      when :add then
        puts "#{value} + #{value} = #{value + value}"
      else 
        puts "Don't know what to do with message : #{message}"
    end
  end
end

myActor = Actors.actor_of(MyActor).start

myActor | [:mult, 4]
myActor | [:add, 4]
myActor | [:something, 4]

sleep 2

myActor.stop
